packages/java-edition/src/mcfunction/node/argument.ts

```diff
@@ -304,6 +349,61 @@ export namespace ItemNode {
 	}
 }
 
+export interface ComponentListNode extends core.AstNode {
+	type: 'mcfunction:component_list'
+	children: core.PairNode<core.ResourceLocationNode, nbt.NbtNode>[]
+}
+
+export namespace ComponentListNode {
+	export function is(node: core.AstNode): node is ComponentListNode {
+		return (node as ComponentListNode).type === 'mcfunction:component_list'
+	}
+}
+
+export interface ComponentPredicatesNode extends core.AstNode {
+	type: 'mcfunction:component_predicates'
+	children: ComponentTestBaseNode[]
+}
+
+export namespace ComponentPredicatesNode {
+	export function is(node: core.AstNode): node is ComponentPredicatesNode {
+		return (node as ComponentPredicatesNode).type ===
+			'mcfunction:component_predicates'
+	}
+}
+
+export interface ComponentTestBaseNode extends core.AstNode {
+	negated: boolean
+}
+
+export namespace ComponentTestBaseNode {
+	export function is(node: core.AstNode): node is ComponentTestBaseNode {
+		return (node as ComponentTestBaseNode).type.startsWith(
+			'mcfunction:component_test',
+		)
+	}
+}
+
+export interface ComponentTestExactNode extends ComponentTestBaseNode {
+	type: 'mcfunction:component_test_exact'
+	children: [core.ResourceLocationNode, nbt.NbtNode]
+	component: core.ResourceLocationNode
+	value: nbt.NbtNode
+}
+
+export interface ComponentTestExistsNode extends ComponentTestBaseNode {
+	type: 'mcfunction:component_test_exists'
+	children: [core.ResourceLocationNode]
+	component: core.ResourceLocationNode
+}
+
+export interface ComponentTestSubpredicateNode extends ComponentTestBaseNode {
+	type: 'mcfunction:component_test_subpredicate'
```

Comment:
(minor) "sub predicate" are two words

```suggestion
export interface ComponentTestSubPredicateNode extends ComponentTestBaseNode {
	type: 'mcfunction:component_test_sub_predicate'
```